app/controllers/admin/site_settings_controller.rb

```diff
@@ -16,6 +16,10 @@ def update
     value.strip! if value.is_a?(String)
     raise_access_hidden_setting(id)
 
+    if SiteSetting.type_supervisor.get_type(id) == :uploaded_image_list
+      value = value.split("|").map { |url| Upload.get_from_url(url) } || []
```

Comment:
We need to be careful about this as this will generate at least 1 request per upload in the list.

Reply:
What do you mean by "1 request per upload"? You are correct, it performs one database lookup (`find_by`) for each upload.

I can make it a bulk operation if I reimplement `Upload#get_from_url` (pretty much `where` and not `find_by` and match regex with all inputs).

Reply:
Exactly. If there are 10 uploads in the list, we'll be doing at least 10 SQL queries every time we're loading that site setting...